x-pack/plugins/gis/public/elasticsearch_geo_utils.js

```diff
@@ -231,47 +215,41 @@ export function convertMapExtentToEnvelope({ maxLat, maxLon, minLat, minLon }) {
   }
 
   if (maxLon > 180) {
-    // bounds cross datleine east to west, slit into 2 shapes
+    // bounds cross dateline east to west
     const overlapWestOfDateLine = maxLon - 180;
-    return [
-      convertMapExtentToEnvelope({
-        maxLat,
-        maxLon: 180,
-        minLat,
-        minLon,
-      }),
-      convertMapExtentToEnvelope({
-        maxLat,
-        maxLon: -180 + overlapWestOfDateLine,
-        minLat,
-        minLon: -180,
-      }),
-    ];
+    return convertMapExtentToPolygon({
+      maxLat,
+      maxLon: -180 + overlapWestOfDateLine,
+      minLat,
+      minLon,
+    });
   }
 
   if (minLon < -180) {
-    // bounds cross datleine west to east, slit into 2 shapes
+    // bounds cross dateline west to east, slit into 2 shapes
     const overlapEastOfDateLine = Math.abs(minLon) - 180;
-    return [
-      convertMapExtentToEnvelope({
-        maxLat,
-        maxLon: 180,
-        minLat,
-        minLon: 180 - overlapEastOfDateLine,
-      }),
-      convertMapExtentToEnvelope({
-        maxLat,
-        maxLon: maxLon,
-        minLat,
-        minLon: -180,
-      }),
-    ];
+    return convertMapExtentToPolygon({
+      maxLat,
+      maxLon,
+      minLat,
+      minLon: 180 - overlapEastOfDateLine,
+    });
   }
 
+  // GeoJSON mandates that the outer polygon must be counterclockwise to avoid ambiguous polygons
+  // when the shape crosses the dateline
+  const left = minLon;
+  const right = maxLon;
+  const top = maxLat > 90 ? 90 : maxLat;
```

Comment:
could you extract this clamping function in util-module? we do clamping here too  https://github.com/thomasneirynck/kibana/blob/ffc8bae820926216e6e11ebe4e9a113b94ea4aa5/x-pack/plugins/gis/public/shared/layers/sources/es_geohashgrid_source/convert_to_geojson.js#L118, and we'll do it for the bounds fitting as well.

Reply:
`clampGrid` from `convert_to_geojson` is not clamping for dateline crossing. This is hear to clamp the geoCentroid to be within the geohash grid. The reason is that a single matching point in an array will match a document to a grid. In some cases, all of the other points in that grid may be way far away from the grid. This would push the centroid out of the grid. This function just ensures that the centroid will always be in the grid for this weird case